File: lib/custum widgets/custom_app_bar/custom_app_bar.dart
// lib/widgets/custom_app_bar.dart
import 'package:flutter/material.dart';
import '../colors/colors.dart';

class CustomAppBar extends StatelessWidget implements PreferredSizeWidget {
  final String hospitalName;
  final String department;
  final int notificationCount;
  final int patientCount;
  final int appointmentCount;
  final String? profileImageUrl;
  final VoidCallback? onNotificationsPressed;
  final VoidCallback? onProfilePressed;
  final VoidCallback? onDrawerPressed;

  const CustomAppBar({
    Key? key,
    this.hospitalName = "MediCare Central",
    this.department = "Administration",
    this.notificationCount = 3,
    this.patientCount = 24,
    this.appointmentCount = 8,
    this.profileImageUrl,
    this.onNotificationsPressed,
    this.onProfilePressed,
    this.onDrawerPressed,
  }) : super(key: key);

  @override
  Size get preferredSize => const Size.fromHeight(90);

  @override
  Widget build(BuildContext context) {
    final mediaQuery = MediaQuery.of(context);
    final size = mediaQuery.size;
    final isLargeScreen = size.width > 1200;
    final isMediumScreen = size.width > 768;
    final isSmallScreen = size.width <= 768;

    final now = DateTime.now();
    final formattedDate = '${_getWeekday(now.weekday)} ${now.day} ${_getMonth(now.month)}';

    return AppBar(
      backgroundColor: Colors.transparent,
      elevation: 0,
      automaticallyImplyLeading: false,
      toolbarHeight: preferredSize.height,
      flexibleSpace: Container(
        decoration: BoxDecoration(
          color: Color(0xFF109A8A), // Primary color
          borderRadius: BorderRadius.only(
            bottomLeft: Radius.circular(16),
            bottomRight: Radius.circular(16),
          ),
          boxShadow: [
            BoxShadow(
              color: Color(0xFF2EC1B1).withOpacity(0.4),
              blurRadius: 20,
              offset: Offset(0, 8),
            ),
          ],
        ),
        child: SafeArea(
          bottom: false,
          child: Padding(
            padding: EdgeInsets.symmetric(horizontal: isLargeScreen ? 32 : isMediumScreen ? 24 : 16),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              crossAxisAlignment: CrossAxisAlignment.center,
              children: [
                // Left: Drawer Button and Hospital Info
                Row(
                  children: [
                    // Drawer Button
                    IconButton(
                      onPressed: onDrawerPressed ?? () {
                        Scaffold.of(context).openDrawer();
                      },
                      icon: Icon(
                        Icons.menu,
                        color: Colors.white,
                        size: isLargeScreen ? 28 : isMediumScreen ? 26 : 24,
                      ),
                    ),

                    SizedBox(width: isMediumScreen ? 20 : 16),

                    // Hospital Info
                    Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          hospitalName,
                          style: TextStyle(
                            fontSize: isLargeScreen ? 22 : isMediumScreen ? 20 : 18,
                            fontWeight: FontWeight.w700,
                            color: Colors.white,
                            letterSpacing: -0.3,
                          ),
                        ),
                        SizedBox(height: 4),
                        Container(
                          padding: EdgeInsets.symmetric(
                            horizontal: isMediumScreen ? 16 : 12,
                            vertical: 6,
                          ),
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.2),
                            borderRadius: BorderRadius.circular(20),
                            border: Border.all(
                              color: Colors.white.withOpacity(0.3),
                              width: 1,
                            ),
                          ),
                          child: Text(
                            department.toUpperCase(),
                            style: TextStyle(
                              fontSize: isMediumScreen ? 12 : 11,
                              fontWeight: FontWeight.w600,
                              color: Colors.white,
                              letterSpacing: 0.5,
                            ),
                          ),
                        ),
                      ],
                    ),
                  ],
                ),

                // Center: Quick Stats (Hidden on small screens)
                if (isMediumScreen)
                  Expanded(
                    child: Center(
                      child: Container(
                        padding: EdgeInsets.symmetric(horizontal: 24, vertical: 10),
                        decoration: BoxDecoration(
                          color: Colors.white.withOpacity(0.15),
                          borderRadius: BorderRadius.circular(20),
                          border: Border.all(
                            color: Colors.white.withOpacity(0.2),
                            width: 1,
                          ),
                        ),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            _buildStatItem(
                              icon: Icons.people,
                              value: patientCount.toString(),
                              label: 'Patients',
                              iconColor: Colors.white,
                            ),
                            SizedBox(width: 24),
                            Container(
                              width: 1,
                              height: 30,
                              color: Colors.white.withOpacity(0.3),
                            ),
                            SizedBox(width: 24),
                            _buildStatItem(
                              icon: Icons.calendar_today,
                              value: appointmentCount.toString(),
                              label: 'Appointments',
                              iconColor: Colors.white,
                            ),
                            SizedBox(width: 24),
                            Container(
                              width: 1,
                              height: 30,
                              color: Colors.white.withOpacity(0.3),
                            ),
                            SizedBox(width: 24),
                            _buildStatItem(
                              icon: Icons.today,
                              value: formattedDate,
                              label: 'Today',
                              iconColor: Colors.white,
                              isDate: true,
                            ),
                          ],
                        ),
                      ),
                    ),
                  ),

                // Right: User Actions
                Row(
                  children: [
                    // Notifications
                    Stack(
                      clipBehavior: Clip.none,
                      children: [
                        IconButton(
                          onPressed: onNotificationsPressed ?? () {},
                          icon: Icon(
                            Icons.notifications_outlined,
                            color: Colors.white,
                            size: isLargeScreen ? 26 : isMediumScreen ? 24 : 22,
                          ),
                        ),
                        if (notificationCount > 0)
                          Positioned(
                            right: 8,
                            top: 8,
                            child: Container(
                              padding: EdgeInsets.all(2),
                              constraints: BoxConstraints(
                                minWidth: 18,
                                minHeight: 18,
                              ),
                              decoration: BoxDecoration(
                                color: Colors.red,
                                shape: BoxShape.circle,
                                border: Border.all(
                                  color: Colors.white,
                                  width: 2,
                                ),
                              ),
                              child: Center(
                                child: Text(
                                  notificationCount > 9 ? '9+' : '$notificationCount',
                                  style: TextStyle(
                                    fontSize: 9,
                                    fontWeight: FontWeight.bold,
                                    color: Colors.white,
                                  ),
                                ),
                              ),
                            ),
                          ),
                      ],
                    ),

                    SizedBox(width: isMediumScreen ? 8 : 4),

                    // User Profile
                    Container(
                      width: isLargeScreen ? 50 : isMediumScreen ? 46 : 42,
                      height: isLargeScreen ? 50 : isMediumScreen ? 46 : 42,
                      decoration: BoxDecoration(
                        shape: BoxShape.circle,
                        border: Border.all(
                          color: Colors.white,
                          width: 2,
                        ),
                        boxShadow: [
                          BoxShadow(
                            color: Colors.black.withOpacity(0.1),
                            blurRadius: 8,
                            offset: Offset(0, 4),
                          ),
                        ],
                      ),
                      child: ClipOval(
                        child: profileImageUrl != null
                            ? Image.network(
                          profileImageUrl!,
                          fit: BoxFit.cover,
                          errorBuilder: (context, error, stackTrace) =>
                              _buildFallbackAvatar(),
                        )
                            : _buildFallbackAvatar(),
                      ),
                    ).asClickable(onProfilePressed ?? () {}),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildFallbackAvatar() {
    return Container(
      color: Color(0xFF2EC1B1).withOpacity(0.8),
      child: Center(
        child: Icon(
          Icons.person,
          color: Colors.white,
          size: 24,
        ),
      ),
    );
  }

  Widget _buildStatItem({
    required IconData icon,
    required String value,
    required String label,
    required Color iconColor,
    bool isDate = false,
  }) {
    return Row(
      children: [
        Icon(
          icon,
          color: iconColor,
          size: 20,
        ),
        SizedBox(width: 8),
        Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              value,
              style: TextStyle(
                fontSize: isDate ? 14 : 16,
                fontWeight: FontWeight.w700,
                color: Colors.white,
              ),
            ),
            Text(
              label,
              style: TextStyle(
                fontSize: 11,
                color: Colors.white.withOpacity(0.9),
                fontWeight: FontWeight.w500,
              ),
            ),
          ],
        ),
      ],
    );
  }

  String _getWeekday(int weekday) {
    switch (weekday) {
      case 1: return 'Mon';
      case 2: return 'Tue';
      case 3: return 'Wed';
      case 4: return 'Thu';
      case 5: return 'Fri';
      case 6: return 'Sat';
      case 7: return 'Sun';
      default: return '';
    }
  }

  String _getMonth(int month) {
    switch (month) {
      case 1: return 'Jan';
      case 2: return 'Feb';
      case 3: return 'Mar';
      case 4: return 'Apr';
      case 5: return 'May';
      case 6: return 'Jun';
      case 7: return 'Jul';
      case 8: return 'Aug';
      case 9: return 'Sep';
      case 10: return 'Oct';
      case 11: return 'Nov';
      case 12: return 'Dec';
      default: return '';
    }
  }
}

// Extension for clickable widgets with ripple effect
extension ClickableExtension on Widget {
  Widget asClickable(VoidCallback onTap) {
    return Material(
      color: Colors.transparent,
      child: InkWell(
        onTap: onTap,
        borderRadius: BorderRadius.circular(50),
        child: this,
      ),
    );
  }
}

File: lib/custum widgets/custom_app_bar/CustomAppBar.java
// lib/widgets/custom_app_bar.dart
import 'package:flutter/material.dart';
import '../colors/colors.dart';

class CustomAppBar extends StatelessWidget implements PreferredSizeWidget {
  final bool isTablet;
  final String hospitalName;
  final String department;
  final int notificationCount;

  const CustomAppBar({
    Key? key,
    required this.isTablet,
    this.hospitalName = "MediCare Central",
    this.department = "Administration",
    this.notificationCount = 3,
  }) : super(key: key);

  @override
  Size get preferredSize => Size.fromHeight(isTablet ? 100 : 85);

  @override
  Widget build(BuildContext context) {
    final timeOfDay = _getTimeOfDay();
    
    return AppBar(
      backgroundColor: Colors.transparent,
      elevation: 0,
      automaticallyImplyLeading: false,
      flexibleSpace: Container(
        decoration: BoxDecoration(
          gradient: LinearGradient(
            colors: [
              AppColors.primaryColor, // Deep blue
              Color(0xFF0052D4), // Medical blue
              AppColors.accentColor.withOpacity(0.9), // Teal accent
            ],
            begin: Alignment.topLeft,
            end: Alignment.bottomRight,
            stops: [0.0, 0.6, 1.0],
          ),
          borderRadius: const BorderRadius.only(
            bottomLeft: Radius.circular(30),
            bottomRight: Radius.circular(30),
          ),
          boxShadow: [
            BoxShadow(
              color: AppColors.primaryColor.withOpacity(0.3),
              blurRadius: 15,
              spreadRadius: 2,
              offset: Offset(0, 4),
            ),
          ],
        ),
        child: SafeArea(
          child: Padding(
            padding: EdgeInsets.symmetric(horizontal: isTablet ? 28 : 18),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                // Left side: Hospital Logo and Info
                Row(
                  children: [
                    // Hospital Logo/Icon
                    Container(
                      width: isTablet ? 52 : 44,
                      height: isTablet ? 52 : 44,
                      decoration: BoxDecoration(
                        color: Colors.white.withOpacity(0.15),
                        borderRadius: BorderRadius.circular(16),
                        border: Border.all(
                          color: Colors.white.withOpacity(0.3),
                          width: 1.5,
                        ),
                      ),
                      child: Icon(
                        Icons.medical_services_outlined,
                        color: Colors.white,
                        size: isTablet ? 28 : 24,
                      ),
                    ),
                    SizedBox(width: isTablet ? 16 : 12),
                    
                    // Hospital Info
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      mainAxisAlignment: MainAxisAlignment.center,
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Text(
                          hospitalName,
                          style: TextStyle(
                            fontSize: isTablet ? 18 : 14,
                            fontWeight: FontWeight.w700,
                            color: Colors.white,
                            letterSpacing: -0.3,
                          ),
                        ),
                        const SizedBox(height: 2),
                        Container(
                          padding: EdgeInsets.symmetric(
                            horizontal: isTablet ? 10 : 8,
                            vertical: 2,
                          ),
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.2),
                            borderRadius: BorderRadius.circular(20),
                          ),
                          child: Text(
                            department,
                            style: TextStyle(
                              fontSize: isTablet ? 13 : 11,
                              color: Colors.white.withOpacity(0.9),
                              fontWeight: FontWeight.w500,
                            ),
                          ),
                        ),
                      ],
                    ),
                  ],
                ),

                // Center: Welcome Message
                Expanded(
                  child: Padding(
                    padding: EdgeInsets.symmetric(horizontal: isTablet ? 24 : 16),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      crossAxisAlignment: CrossAxisAlignment.center,
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Icon(
                              _getTimeOfDayIcon(timeOfDay),
                              color: Colors.white.withOpacity(0.9),
                              size: isTablet ? 18 : 16,
                            ),
                            SizedBox(width: 6),
                            Text(
                              '$timeOfDay, ',
                              style: TextStyle(
                                fontSize: isTablet ? 15 : 13,
                                color: Colors.white.withOpacity(0.9),
                                fontWeight: FontWeight.w500,
                              ),
                            ),
                            Text(
                              'Dr. Anderson',
                              style: TextStyle(
                                fontSize: isTablet ? 15 : 13,
                                fontWeight: FontWeight.w700,
                                color: Colors.white,
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 4),
                        Text(
                          _getWelcomeMessage(timeOfDay),
                          style: TextStyle(
                            fontSize: isTablet ? 12 : 10,
                            color: Colors.white.withOpacity(0.8),
                            fontWeight: FontWeight.w400,
                          ),
                        ),
                      ],
                    ),
                  ),
                ),

                // Right side: Actions
                Row(
                  children: [
                    // Emergency Button
                    Container(
                      margin: EdgeInsets.only(right: isTablet ? 12 : 8),
                      child: Material(
                        color: Colors.transparent,
                        child: InkWell(
                          onTap: () {
                            // Handle emergency
                          },
                          borderRadius: BorderRadius.circular(20),
                          child: Container(
                            padding: EdgeInsets.symmetric(
                              horizontal: isTablet ? 16 : 12,
                              vertical: 8,
                            ),
                            decoration: BoxDecoration(
                              color: Colors.red.withOpacity(0.15),
                              borderRadius: BorderRadius.circular(20),
                              border: Border.all(
                                color: Colors.red.withOpacity(0.3),
                                width: 1.5,
                              ),
                            ),
                            child: Row(
                              children: [
                                Icon(
                                  Icons.emergency_outlined,
                                  color: Colors.white,
                                  size: isTablet ? 16 : 14,
                                ),
                                SizedBox(width: 6),
                                Text(
                                  'EMERGENCY',
                                  style: TextStyle(
                                    fontSize: isTablet ? 12 : 10,
                                    fontWeight: FontWeight.w700,
                                    color: Colors.white,
                                    letterSpacing: 0.5,
                                  ),
                                ),
                              ],
                            ),
                          ),
                        ),
                      ),
                    ),

                    // Notifications with Badge
                    Stack(
                      children: [
                        Container(
                          width: isTablet ? 46 : 40,
                          height: isTablet ? 46 : 40,
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.15),
                            shape: BoxShape.circle,
                            border: Border.all(
                              color: Colors.white.withOpacity(0.3),
                              width: 1.5,
                            ),
                          ),
                          child: IconButton(
                            onPressed: () {},
                            icon: Icon(
                              Icons.notifications_outlined,
                              color: Colors.white,
                              size: isTablet ? 22 : 20,
                            ),
                          ),
                        ),
                        if (notificationCount > 0)
                          Positioned(
                            right: isTablet ? 6 : 4,
                            top: isTablet ? 6 : 4,
                            child: Container(
                              width: isTablet ? 20 : 18,
                              height: isTablet ? 20 : 18,
                              decoration: BoxDecoration(
                                color: Colors.red,
                                shape: BoxShape.circle,
                                border: Border.all(
                                  color: Colors.white,
                                  width: 2,
                                ),
                              ),
                              child: Center(
                                child: Text(
                                  notificationCount > 9 ? '9+' : '$notificationCount',
                                  style: TextStyle(
                                    fontSize: isTablet ? 10 : 8,
                                    fontWeight: FontWeight.bold,
                                    color: Colors.white,
                                  ),
                                ),
                              ),
                            ),
                          ),
                      ],
                    ),

                    SizedBox(width: isTablet ? 12 : 8),

                    // Profile with Status Indicator
                    Stack(
                      children: [
                        Container(
                          width: isTablet ? 52 : 44,
                          height: isTablet ? 52 : 44,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            border: Border.all(
                              color: Colors.white,
                              width: 2.5,
                            ),
                            boxShadow: [
                              BoxShadow(
                                color: Colors.black.withOpacity(0.2),
                                blurRadius: 8,
                                spreadRadius: 1,
                              ),
                            ],
                          ),
                          child: ClipOval(
                            child: Image.network(
                              'https://images.unsplash.com/photo-1559839734-2b71ea197ec2?w=200&h=200&fit=crop',
                              fit: BoxFit.cover,
                            ),
                          ),
                        ),
                        Positioned(
                          bottom: 0,
                          right: 0,
                          child: Container(
                            width: isTablet ? 14 : 12,
                            height: isTablet ? 14 : 12,
                            decoration: BoxDecoration(
                              color: Colors.green,
                              shape: BoxShape.circle,
                              border: Border.all(
                                color: Colors.white,
                                width: 2,
                              ),
                            ),
                          ),
                        ),
                      ],
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  String _getTimeOfDay() {
    final hour = DateTime.now().hour;
    if (hour < 12) return 'Morning';
    if (hour < 17) return 'Afternoon';
    return 'Evening';
  }

  IconData _getTimeOfDayIcon(String timeOfDay) {
    switch (timeOfDay) {
      case 'Morning':
        return Icons.wb_sunny_outlined;
      case 'Afternoon':
        return Icons.light_mode_outlined;
      case 'Evening':
        return Icons.nights_stay_outlined;
      default:
        return Icons.access_time_outlined;
    }
  }

  String _getWelcomeMessage(String timeOfDay) {
    switch (timeOfDay) {
      case 'Morning':
        return 'Ready for today\'s patients';
      case 'Afternoon':
        return 'Keep up the great work';
      case 'Evening':
        return 'Night shift activated';
      default:
        return 'Hospital Dashboard';
    }
  }
}